fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/RuntimeFilterTranslator.java

```diff
@@ -129,7 +130,15 @@ public void createLegacyRuntimeFilter(RuntimeFilter filter, JoinNodeBase node, P
             if (!src.getType().equals(targetExpr.getType()) && filter.getType() != TRuntimeFilterType.BITMAP) {
                 targetExpr = new CastExpr(src.getType(), targetExpr);
             }
-            SlotRef targetSlot = targetSlotRef.getSrcSlotRef();
+            SlotRef targetSlot;
+            if (filter.getTargetScans().get(i) instanceof PhysicalCTEConsumer) {
+                targetSlot = targetSlotRef.unwrapSlotRef();
+            } else {
+                targetSlot = targetSlotRef.getSrcSlotRef();
```

Comment:
could we compute src slot in Nereids' rules to avoid compute here?

Reply:
targetSlotRef is original planner's SlotRef, not slotreference